itou/external_data/apis/pe_connect.py

```diff
@@ -0,0 +1,280 @@
+    Does not return country related fields (only France ATM)
+
+    See: https://www.emploi-store-dev.fr/portail-developpeur-cms/home/catalogue-des-api/documentation-des-api/api/api-pole-emploi-connect/api-peconnect-coordonnees-v1.html
+    """
+    keys = ["adresse1", "adresse2", "adresse3", "adresse4", "codePostal", "codeINSEE", "libelleCommune"]
+    return _fields_or_failed(_call_api(ESD_COORDS_API, token), keys)
+
+
+def _get_compensations(token):
+    """
+    Get user "compensations" (social allowance):
+
+    Return a dict with fields:
+        * beneficiairePrestationSolidarite (has one or more of AER, AAH, ASS, RSA)
+        * beneficiaireAssuranceChomage (has ARE or ASP)
+
+    See: https://www.emploi-store-dev.fr/portail-developpeur-cms/home/catalogue-des-api/documentation-des-api/api/api-pole-emploi-connect/api-indemnisations-v1.html
+    """
+    keys = ["beneficiairePrestationSolidarite", "beneficiaireAssuranceChomage"]
+    return _fields_or_failed(_call_api(ESD_COMPENSATION_API, token), keys)
+
+
+def _get_aggregated_user_data(token):
+    """
+    Aggregates all needed user data before formatting and storage.
+    Returns a pair status and a "flat" dict.
+    """
+    # Include API results "à volonté"
+    results = [
+        _get_userinfo(token),
+        _get_birthdate(token),
+        _get_status(token),
+        _get_address(token),
+        _get_compensations(token),
+    ]
+
+    ok = all(results)
+    partial = not ok and any(results)
+    cleaned_results = [part for part in results if part]
+    resp = dict()
+
+    if ok:
+        status = ExternalDataImport.STATUS_OK
+    elif partial:
+        status = ExternalDataImport.STATUS_PARTIAL
+    else:
+        status = ExternalDataImport.STATUS_FAILED
+
+    for result in cleaned_results:
+        resp = {**resp, **result}
+
+    return status, resp
+
+
+# External user data from PE Connect API:
+# * transform raw data from API
+# * dispatch data into models if possible
+# * or store as key / value if needed
+
+
+def _store_user_data(user, status, data):
+    # Set a trace of data import, whatever the resulting status
+    data_import = ExternalDataImport.objects.pe_import_for_user(user).first()
+
+    # User part:
+    # Can be directly "inserted" in to the model
+
+    # Birth date
+    if data.get("dateDeNaissance"):
+        user.birthdate = user.birthdate or parse_datetime(data.get("dateDeNaissance"))
+
+    # User address
+    if data.get("adresse4"):
+        user.address_line_1 = "" or user.address_line_1 or data.get("adresse4")
+
+    if data.get("adresse2"):
+        user.address_line_2 = "" or user.address_line_2 or data.get("adresse2")
+
+    if data.get("codePostal"):
+        user.post_code = user.post_code or data.get("codePostal")
+
+    if data.get("libelleCommune"):
+        user.city = user.city or data.get("libelleCommune")
+
+    user.save()
+
+    # The following will be stored as k/v
+    external_user_data = []
+    result_keys = list(data.keys())
```

Comment:
Pas besoin de transformer `data.keys()` explicitement en `list()` ici, [le `in` d'après fonctionnera quand même](https://docs.python.org/3/library/stdtypes.html#dict-views).